pkg/scheduler/framework/plugins/clusteraffinity/types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clusteraffinity
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+// AffinityTerm is a processed version of ClusterSelectorTerm.
+type AffinityTerm struct {
+	selector labels.Selector
+}
+
+// Matches returns true if the cluster matches the label selector.
+func (at *AffinityTerm) Matches(cluster *fleetv1beta1.MemberCluster) bool {
+	return at.selector.Matches(labels.Set(cluster.Labels))
+}
+
+// preferredAffinityTerm is a "processed" representation of PreferredClusterSelector.
+type preferredAffinityTerm struct {
+	AffinityTerm
+	weight int32
+}
+
+// PreferredAffinityTerms is a "processed" representation of []PreferredClusterSelector.
+type PreferredAffinityTerms struct {
```

Comment:
A nit here: it can just be a type alias I assume?

Reply:
it has to be a struct, otherwise, cannot compute the score of these terms.

```
func (t *PreferredAffinityTerms) Score(cluster *fleetv1beta1.MemberCluster) int32 {
	var score int32
	for _, term := range t.terms {
		if term.AffinityTerm.Matches(cluster) {
			score += term.weight
		}
	}
	return score
}
```

Reply:
Why? You can still define methods for aliases

Reply:
https://go.dev/play/p/FUbp0RsFNn7

Reply:
This isn't a blocking thing in any way though, I do not have any strong opposition on this front.